Update TinkerPop to 3.5.3

FlorianHockmann opened the pull request:
Fixes #2997

The code changes where necessary because of TINKERPOP-2605 which allows `null` as an argument in more places.
The changes to `FulgoraMemory` are basically the same changes that were done in TinkerPop for `TinkerMemory`:
https://github.com/apache/tinkerpop/pull/1471/files#diff-876c2aa792d2cde263913b85027dd772050f9b99bb0a2d53ad8d8be38e073f13

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

FlorianHockmann commented:
### Short status update

Some TinkerPop tests are still failing here. I could fix the OLAP specific tests already, so now only these tests are still failing:

* `g_E_hasXnullX`
* `g_V_hasXnullX`

which both produce a NPE because we're trying to create a `HasContainer` with a `null` `key` which isn't valid right now: https://issues.apache.org/jira/browse/TINKERPOP-2727

I hope that the tests pass when TINKERPOP-2727 is resolved.

FlorianHockmann commented:
This is now ready for review. The only still failing test is also failing on `master`, see #3017. I'll update the PR description for the current state.

farodin91 commented on janusgraph-core/src/main/java/org/janusgraph/graphdb/query/vertex/BaseVertexCentricQueryBuilder.java:
```diff
@@ -179,7 +179,10 @@ public Q type(RelationType type) {
         if (types == null) types = NO_TYPES;
-        for (String type : types) Preconditions.checkArgument(StringUtils.isNotBlank(type), "Invalid type: %s", type);
+        for (String type : types)
+            if (type != null ) {
```
```suggestion
            if (type != null) {
```

farodin91 reviewed (approved):
LGTM

porunov reviewed (commented):
Thank you @FlorianHockmann ! LGTM but I have only one question about the new added file.

What is the purpose of this file and what is it (tinkerpop-modern-v3d0.kryo)?

https://github.com/JanusGraph/janusgraph/blob/89bafc5529bd26960efba3055abf598d31ca0bf8/janusgraph-inmemory/file:/home/user/.m2/repository/org/janusgraph/janusgraph-core/0.6.2-SNAPSHOT/test-case-data/StandardJanusGraph/resources/tinkerpop-modern-v3d0.kryo

FlorianHockmann commented:
> What is the purpose of this file and what is it (tinkerpop-modern-v3d0.kryo)?

Oh, I've added that by accident. Thanks for noticing! I guess it was created as part of the build process somehow, maybe when executing the TinkerPop tests, but I'm not sure.

But I just updated the commit to remove this file and also resolved the nit from @farodin91.

porunov reviewed (approved):
LGTM. Thank you @FlorianHockmann !
